### PR TITLE
 Update embedded LDAP to 2.0.0-AM25

### LINKED ISSUE #5359: Update embedded LDAP to Apache DS 2.0.0-AM25
The current LDAP version 1.5.7 is not supported in Java 8. To make the product supported in java 11. we want to update the embedded LDAP version to 2.0.0-AM25

### PR DESCRIPTION
**Pull request by senthalan**
### Proposed changes in this pull request

The current LDAP version 1.5.7 is not supported in Java 8. To make the product supported in java 11. we want to update the embedded LDAP version to 2.0.0-AM25. Fix https://github.com/wso2/product-is/issues/5359

### CODE REVIEW

File: components/org.wso2.carbon.ldap.server/src/main/java/org/wso2/carbon/apacheds/impl/ApacheLDAPServer.java [hunk 1]
```diff
@@ -282,20 +279,25 @@ public void changeConnectionUserPassword(String password)
                     passwordToStore = passwordToStore + hash;
 
                     adminPrincipal.setUserPassword(passwordToStore.getBytes());
+                    AttributesFactory attributesFactory = new AttributesFactory();
+                    Attribute passwordAttribute = new DefaultAttribute(getAttributeType("userPassword"));
+                    try {
+                        passwordAttribute.add(passwordToStore.getBytes());
+                    } catch (LdapInvalidAttributeValueException e) {
+                        String msg = "Failed .";
```
Comment (mefarazath) [inline components/org.wso2.carbon.ldap.server/src/main/java/org/wso2/carbon/apacheds/impl/ApacheLDAPServer.java]: Let's have a meaningful error message here

File: components/org.wso2.carbon.ldap.server/src/main/java/org/wso2/carbon/apacheds/impl/CarbonDirectoryServiceFactory.java [hunk 2]
```diff
@@ -18,31 +18,38 @@
...
 import org.wso2.carbon.ldap.server.exception.DirectoryServerException;
 
 import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
-class CarbonDirectoryServiceFactory implements DirectoryServiceFactory {
+class CarbonDirectoryServiceFactory {
```
Comment (mefarazath) [inline components/org.wso2.carbon.ldap.server/src/main/java/org/wso2/carbon/apacheds/impl/CarbonDirectoryServiceFactory.java:52]: has the interface DirectoryServiceFactory been removed?

**Review by mefarazath — COMMENTED** [review]
Added few minor comments.

File: components/org.wso2.carbon.ldap.server/src/main/java/org/wso2/carbon/apacheds/impl/CarbonDirectoryServiceFactory.java (hunk 2 above)
Comment (senthalan) [reply components/org.wso2.carbon.ldap.server/src/main/java/org/wso2/carbon/apacheds/impl/CarbonDirectoryServiceFactory.java:52]: yes, it got moved to the core-annotation modules which apache ds use for testing. So we decided to remove it. Having that dependency needs test dependency in the run time. refer to https://github.com/apache/directory-server/blob/master/core-annotations/pom.xml

**Review by darshanasbg — APPROVED** [review]

**Review by mefarazath — APPROVED** [review]

**Pull request merged**